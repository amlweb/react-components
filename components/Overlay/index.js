import React from 'react';
import PropTypes from 'prop-types';

export default function Overlay({ noIcon, showOverlay }) {
  const classes = ['overlay', 'animating', 'react-hidden'];

  if (noIcon) {
    classes.push('no-icon');
  }
  if (showOverlay) {
    classes.push('react-hidden--show');
  }

  return (
    <div className={classes.join(' ')}>
      <span className="overlay__icon icon-spinner6" />
    </div>
  );
}

Overlay.defaultProps = {
  noIcon: false,
};

Overlay.propTypes = {
  showOverlay: PropTypes.bool.isRequired,
  noIcon: PropTypes.bool,
};
