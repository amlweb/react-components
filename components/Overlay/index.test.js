import React from 'react';
import renderer from 'react-test-renderer';
import Overlay from '../Overlay';

describe('Showing or hiding overlay', () => {
  test('Show overlay', () => {
    const component = renderer.create(<Overlay showOverlay />);

    expect(component.toJSON()).toMatchSnapshot();
  });

  test('Hide overlay', () => {
    const component = renderer.create(<Overlay showOverlay={false} />);

    expect(component.toJSON()).toMatchSnapshot();
  });

  test('Show overlay with no-icon class', () => {
    const component = renderer.create(<Overlay showOverlay noIcon />);

    expect(component.toJSON()).toMatchSnapshot();
  });
});
