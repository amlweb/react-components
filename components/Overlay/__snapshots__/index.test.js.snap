// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Showing or hiding overlay Hide overlay 1`] = `
<div
  className="overlay animating react-hidden"
>
  <span
    className="overlay__icon icon-spinner6"
  />
</div>
`;

exports[`Showing or hiding overlay Show overlay 1`] = `
<div
  className="overlay animating react-hidden react-hidden--show"
>
  <span
    className="overlay__icon icon-spinner6"
  />
</div>
`;

exports[`Showing or hiding overlay Show overlay with no-icon class 1`] = `
<div
  className="overlay animating react-hidden no-icon react-hidden--show"
>
  <span
    className="overlay__icon icon-spinner6"
  />
</div>
`;
